tensorflow/contrib/go/tensor.go

```diff
@@ -0,0 +1,753 @@
+	case DTDouble:
+		auxTensor.DoubleVal = make([]float64, len(dataSer))
+		for i, v := range dataSer {
+			auxTensor.DoubleVal[i] = v.(float64)
+		}
+		dataPtr = reflect.ValueOf(auxTensor.DoubleVal).Pointer()
+	case DTInt8, DTInt16, DTInt32, DTUint8:
+		auxTensor.IntVal = make([]int32, len(dataSer))
+		for i, v := range dataSer {
+			auxTensor.IntVal[i] = int32(reflect.ValueOf(v).Int())
+		}
+		dataPtr = reflect.ValueOf(auxTensor.IntVal).Pointer()
+	case DTInt64:
+		auxTensor.Int64Val = make([]int64, len(dataSer))
+		for i, v := range dataSer {
+			auxTensor.Int64Val[i] = reflect.ValueOf(v).Int()
+		}
+		dataPtr = reflect.ValueOf(auxTensor.Int64Val).Pointer()
+	case DTBool:
+		auxTensor.BoolVal = make([]bool, len(dataSer))
+		for i, v := range dataSer {
+			auxTensor.BoolVal[i] = v.(bool)
+		}
+		dataPtr = reflect.ValueOf(auxTensor.BoolVal).Pointer()
+	case DTString:
+		auxTensor.StringVal = make([][]byte, len(dataSer))
+		for i, v := range dataSer {
+			auxTensor.StringVal[i] = []byte(v.(string))
+		}
+		dataPtr = reflect.ValueOf(auxTensor.StringVal).Pointer()
+	default:
+		return nil, &ErrTensorTypeNotSupported{
+			TensotType: dataType,
+		}
+	}
+
+	tensor, err := newTensor(dataType, ts, unsafe.Pointer(dataPtr), int64(len(dataSer))*dataSize)
+	if err != nil {
+		return nil, err
+	}
+
+	tensor.FloatVal = auxTensor.FloatVal
+	tensor.DoubleVal = auxTensor.DoubleVal
+	tensor.IntVal = auxTensor.IntVal
+	tensor.StringVal = auxTensor.StringVal
+	tensor.ScomplexVal = auxTensor.ScomplexVal
+	tensor.Int64Val = auxTensor.Int64Val
+	tensor.BoolVal = auxTensor.BoolVal
+
+	return tensor, nil
+}
+
+// DataType returns the data type of the elements contained in the tensor.
+func (t *Tensor) DataType() DataType {
+	return DataType(TF_TensorType(t.tensor))
+}
+
+// NumDims returns the number of dimensions in tensor t.
+func (t *Tensor) NumDims() int {
+	return TF_NumDims(t.tensor)
+}
+
+// Shape returns the shape of the tensor.
+func (t *Tensor) Shape() TensorShape {
+	if t.NumDims() == 0 {
+		// This is a scalar tensor
+		return [][]int64{}
+	}
+
+	shape := make([][]int64, t.NumDims())
+	for i := 0; i < t.NumDims(); i++ {
+		shape[i] = []int64{int64(t.Dim(i))}
+	}
+
+	return shape
+}
+
+// Dim returns the size of the specified dimension.
+func (t *Tensor) Dim(n int) int {
+	return int(TF_Dim(t.tensor, n))
+}
+
+// DataSize returns the size of the data in bytes contained in a tensor.
+func (t *Tensor) DataSize() int64 {
+	return TF_TensorByteSize(t.tensor)
+}
+
+// Data returns the data contained in a tensor as bytes slice.
+func (t *Tensor) Data() []byte {
+	length := t.DataSize()
+	return (*[1 << 40]byte)(unsafe.Pointer(TF_TensorData(t.tensor)))[:length:length]
+}
+
+// String returns a human-readable string description of a Tensor.
+func (t *Tensor) String() string {
+	return fmt.Sprintf("%v: dims:%v size:%v", t.DataType(), t.NumDims(), t.DataSize())
+}
+
+// ByteSlices returns the Tensor content as strings slice if the tensor
```

Comment:
doc typo:  as a slice of byte slices (not string)
clarify:  "if the tensor type matches" - if the tensor contains strings